File: _site_local/mod_10_paradigmas_e_padroes_de_projeto/exemplos/oo/PHP/Retangulo/main.php
<?php
// main.php
include 'Retangulo.php';

// Função para capturar a entrada do usuário
function capturarEntrada($mensagem) {
    echo $mensagem;
    return readline(); // Usando readline() para entrada interativa
}

// Solicitando as medidas do Retângulo X
$larguraX = capturarEntrada('Digite a largura do Retângulo X em metros: ');
$alturaX = capturarEntrada('Digite a altura do Retângulo X em metros: ');

// Solicitando as medidas do Retângulo Y
$larguraY = capturarEntrada('Digite a largura do Retângulo Y em metros: ');
$alturaY = capturarEntrada('Digite a altura do Retângulo Y em metros: ');

// Criando os objetos Retângulo X e Y
$retanguloX = new Retangulo($larguraX, $alturaX);
$retanguloY = new Retangulo($larguraY, $alturaY);

// Exibindo as informações dos dois retângulos
echo "\nInformações do Retângulo X:\n";
$retanguloX->exibirInformacoes();

echo "\nInformações do Retângulo Y:\n";
$retanguloY->exibirInformacoes();

// Comparando as áreas dos dois retângulos
echo "\nComparando as áreas dos retângulos:\n";
$retanguloX->compararArea($retanguloY);
?>

File: _site_local/mod_10_paradigmas_e_padroes_de_projeto/exemplos/oo/GO/Retangulo/main.go
// main.go
package main

import (
	"Retangulo/retangulo" // Alterado para o caminho correto do pacote
	"fmt"
)

func main() {
	var larguraX, alturaX, larguraY, alturaY float64

	fmt.Print("Digite a largura do Retângulo X em metros: ")
	fmt.Scanln(&larguraX)
	fmt.Print("Digite a altura do Retângulo X em metros: ")
	fmt.Scanln(&alturaX)

	fmt.Print("Digite a largura do Retângulo Y em metros: ")
	fmt.Scanln(&larguraY)
	fmt.Print("Digite a altura do Retângulo Y em metros: ")
	fmt.Scanln(&alturaY)

	retanguloX := retangulo.Retangulo{Largura: larguraX, Altura: alturaX}
	retanguloY := retangulo.Retangulo{Largura: larguraY, Altura: alturaY}

	fmt.Println("\nInformações do Retângulo X:")
	retanguloX.ExibirInformacoes()

	fmt.Println("\nInformações do Retângulo Y:")
	retanguloY.ExibirInformacoes()

	fmt.Println("\nComparando as áreas dos retângulos:")
	retanguloX.CompararArea(retanguloY)
}
